### PR TITLE
[XPU] add profile

### PR DESCRIPTION
**Pull request by zhupengyang**
## Motivation

> :bulb: If this PR is a Cherry Pick, the PR title needs to follow the format by adding the [Cherry-Pick] label at the very beginning and appending the original PR ID at the end. For example, [Cherry-Pick][CI] Add check trigger and logic(#5191)

> :bulb: 如若此PR是Cherry Pick，PR标题需遵循格式，在最开始加上[Cherry-Pick]标签，以及最后面加上原PR ID，例如[Cherry-Pick][CI] Add check trigger and logic(#5191)

## Checklist

- [ ] Add at least a tag in the PR title.
  - Tag list: [`[FDConfig]`,`[APIServer]`,`[Engine]`, `[Scheduler]`, `[PD Disaggregation]`, `[Executor]`, `[Graph Optimization]`, `[Speculative Decoding]`, `[RL]`, `[Models]`, `[Quantization]`, `[Loader]`, `[OP]`, `[KVCache]`, `[DataProcessor]`, `[BugFix]`, `[Docs]`, `[CI]`, `[Optimization]`, `[Feature]`, `[Benchmark]`, `[Others]`, `[XPU]`, `[HPU]`, `[GCU]`, `[DCU]`, `[Iluvatar]`, `[Metax]`]
  - You can add new tags based on the PR content, but the semantics must be clear.
- [ ] Format your code, run `pre-commit` before commit.
- [ ] Add unit tests. Please write the reason in this PR if no unit tests.
- [ ] Provide accuracy results.
- [ ] If the current PR is submitting to the `release` branch, make sure the PR has been submitted to the `develop` branch, then cherry-pick it to the `release` branch with the `[Cherry-Pick]` PR tag.

### CODE REVIEW

File: fastdeploy/worker/xpu_model_runner.py [hunk 1]
```diff
@@ -1500,6 +1500,120 @@ def profile_run(self) -> None:
...
+    def prefill_only_profile_run(self) -> None:
+        """
+        Execute prefill-only profiling: profile prefill iterations without decode.
+        Triggered by FD_XPU_PROFILE_PREFILL=1.
+        """
+        num_iters = envs.FD_XPU_PROFILE_ITERS
+        batch_size = min(envs.FD_XPU_PROFILE_BATCH_SIZE, self.scheduler_config.max_num_seqs)
+        prefill_len = envs.FD_XPU_PROFILE_PREFILL_LEN
+        prefix_len = envs.FD_XPU_PROFILE_PREFIX_LEN
+        assert (
```
[the remark this answers is not in the document]
Comment (zhupengyang) [reply fastdeploy/worker/xpu_model_runner.py:1512]: 已修改 FD_XPU_PROFILE_BATCH_SIZE 默认值为 1

File: fastdeploy/worker/xpu_model_runner.py [hunk 2]
```diff
@@ -1500,6 +1500,120 @@ def profile_run(self) -> None:
...
+            )
+            self.share_inputs["seq_lens_this_time"][i : i + 1] = tokens_per_iter
+            self.share_inputs["seq_lens_encoder"][i : i + 1] = 0
+            self.share_inputs["step_seq_lens_encoder"][i : i + 1] = 0
+            self.share_inputs["seq_lens_decoder"][i : i + 1] = prefix_len
+            self.share_inputs["step_idx"][i : i + 1] = 0
+            self.share_inputs["max_dec_len"][i : i + 1] = expected_decode_len + 1
+            self.share_inputs["stop_flags"][i : i + 1] = False
+            self.share_inputs["encoder_block_lens"][i : i + 1] = encoder_block_num
+            self.share_inputs["block_tables"][i : i + 1, :block_num] = np.arange(i * block_num, (i + 1) * block_num, 1)
```
[the remark this answers is not in the document]
Comment (zhupengyang) [reply fastdeploy/worker/xpu_model_runner.py:1593]: 已增加强制检查

File: fastdeploy/model_executor/layers/backends/xpu/moe/fused_moe.py [hunk 3]
```diff
@@ -307,6 +308,16 @@ def process_loaded_weights(self, layer: nn.Layer, state_dict):
...
         layer.down_proj_weight.set_value(stacked_down_proj_weights)
 
+    def compute_gate(self, x: paddle.Tensor, gate: nn.Layer) -> paddle.Tensor:
+        """Compute gate output, use random fill in profile mode."""
+        gate_input = x.cast("float32") if gate.weight.dtype == paddle.float32 else x
+        gate_out = gate(gate_input)
+        if gate_out.dtype != paddle.float32:
+            gate_out = gate_out.cast("float32")
+        if envs.FD_XPU_PROFILE_EXPERT_BALANCE:
+            gate_out = paddle.to_tensor(np.random.rand(x.shape[0], gate.weight.shape[1]).astype("float32"))
```
[the remark this answers is not in the document]
Comment (zhupengyang) [reply fastdeploy/model_executor/layers/backends/xpu/moe/fused_moe.py:318]: 不需要。不应该去掉原本的gate计算逻辑

File: fastdeploy/worker/xpu_model_runner.py [hunk 4]
```diff
@@ -1500,6 +1500,126 @@ def profile_run(self) -> None:
...
+    def prefill_only_profile_run(self) -> None:
+        """
+        Execute prefill-only profiling: profile prefill iterations without decode.
+        Triggered by FD_XPU_PROFILE_PREFILL=1.
+        """
+        num_iters = envs.FD_XPU_PROFILE_ITERS
+        batch_size = min(envs.FD_XPU_PROFILE_BATCH_SIZE, self.scheduler_config.max_num_seqs)
+        prefill_len = envs.FD_XPU_PROFILE_PREFILL_LEN
+        prefix_len = envs.FD_XPU_PROFILE_PREFIX_LEN
+        assert (
```
[the remark this answers is not in the document]
Comment (zhupengyang) [reply fastdeploy/worker/xpu_model_runner.py:1512]: 通用处理方式

**Review by hong19860320 — APPROVED** [review]
LGTM

**Review by cmcamdy — APPROVED** [review]

**Review by xyxinyang — APPROVED** [review]
